app/src/main/java/com/example/quizacademytask/StackListFragment.kt

```diff
@@ -55,20 +57,6 @@ class StackListFragment : Fragment(), SimpleAdapter.OnItemClickListener {
         idlingResource = CountingIdlingResource("API")
         binding.toolbarStackList.title = "Topics"
 
-        //DATABASE
-        cardStackDAO = App.db.cardStackDAO()
-        cardDAO = App.db.cardDAO()
-
-        /*Check if course already exists in database. If not then download and insert*/
-        runBlocking {
-            launch {
-                if (model.cardStacks.isEmpty())
-                    getRequest()
```

Comment:
Should/Could I put the API call  in the ViewModel as well?

Reply:
I would use Helper / Utility class for it. Create BackendClient and put your network requests there. Then pass your backend client to your view model as a dependency.